Doc: Further update the parameters about Electronic Structure and OFDFT (#2515)
deepmodeling/abacus-develop: docs/advanced/input_files/input-main.md

- **Type**: String
- **Description**: Choose the basis set.
  - **pw**: Using plane-wave basis set only.

>   - **lcao_in_pw**: Expand the localized atomic set in plane-wave basis.

Reviewer: It is unable in current version, you'd better delete this option in documentation in this PR.

Author: There are still many "lcao_in_pw" in codes, I suggest to delete them in a new pr, and then delete this option.

Reviewer: > There are still many "lcao_in_pw" in codes, I suggest to delete them in a new pr, and then delete this option.

This option is not useless, but can not be used in this version, it can be discussed whether fix or delete.

Author: Is it OK to write as "- **lcao_in_pw**: (Unavailable now) Expand the localized atomic set in plane-wave basis"?

Reviewer: > Is it OK to write as "- **lcao_in_pw**: (Unavailable now) Expand the localized atomic set in plane-wave basis"?

Okay, maybe "Unavailable currently, it will be fixed in future versions".

Author: OK

  - **lcao**: Using localized atomic orbital sets.
- **Default**: pw

### ks_solver

- **Type**: String
- **Description**: Choose the diagonalization methods for the Hamiltonian matrix expanded in a certain basis set.

  For plane-wave basis,

  - **cg**: cg method.
  - **dav**: the Davidson algorithm.

  For atomic orbitals basis,

  - **genelpa**: This method should be used if you choose localized orbitals.
  - **scalapack-gvx**: scalapack can also be used for localized orbitals.

>   - **cusolver**: this method needs building with the cusolver component for lcao and at least one gpu is available.

Reviewer: This option is unable in this version.

Author: If this option is useless, maybe it's better to remove them from the codes, and then delete this option in the doc.

Reviewer: > If this option is useless, maybe it's better to remove them from the codes, and then delete this option in the doc.

This option is not useless, but can not be used in this version, it can be discussed whether fix or delete.

Author: It's done.

  ```

  Then the user has to correct the input file and restart the calculation.

> - **Default**: cg (plane-wave basis), or genelpa (localized atomic orbital basis)

Reviewer: Please use indent for multi default, the correct default for lcao is:
 - genelpa, (compiling option `USE_ELPA` has been set)
 - scalapack_gvx, (compiling option `USE_ELPA` has not been set)

- **Type**: Integer
- **Description**: The number of spin components of wave functions.
  - **1**: Spin degeneracy
  - **2**: Collinear spin polarized.

>   - **4**: For the case of [noncollinear polarized](../scf/spin.md#noncollinear-spin-polarized-calculations), nspin will be automatically set to 4 without being specified in user input.

Reviewer: ```suggestion
  - **4**: For the case of [noncollinear polarized](../scf/spin.md#noncollinear-spin-polarized-calculations), nspin will be automatically set to 4 without being specified by the user.
```

> - **Type**: String

Reviewer: - **Availability**: `smearing_method` is not fixed.

- **Type**: Real
- **Description**: It's the threshold for electronic iteration. It represents the charge density error between two sequential densities from electronic iterations. Usually for local orbitals, usually 1e-6 may be accurate enough.
- **Default**: 1.0e-9 (plane-wave basis), or 1.0e-7 (localized atomic orbital basis).

### scf_thr_type

- **Type**: Int
- **Description**: Choose the calculation method of convergence criterion. 
  - **1**: the criterion is defined as $\Delta\rho_G = \frac{1}{2}\iint{\frac{\Delta\rho(r)\Delta\rho(r')}{|r-r'|}d^3r d^3r'}$.
  - **2**: the criterion is defined as $\Delta\rho_R = \int{|\Delta\rho(r)|d^3r}$.
  
  Note: This parameter is still under testing and the default setting is usually sufficient.

- **Default**: 1 (plane-wave basis), or 2 (localized atomic orbital basis).

### chg_extrap

- **Type**: String

> - **Description**: Methods to do extrapolation of density when ABACUS is doing geometry relaxations.

Reviewer: and molecular dynamics.
